In [1]:
import pandas as pd
import numpy as np

## Original DataFrame

In [2]:
og_data = pd.read_csv("clean_dataset_2(6nov19).csv")

In [3]:
og_data.shape

(424431, 30)

In [4]:
og_data.head()

Vehicle_Symbol  Vehicle_Make_Year  Vehicle_Collision_Coverage_Deductible  \
0              11               2004                                    500   
1              11               1980                                     50   
2              10               2005                                    100   
3              12               2005                                    250   
4              12               2005                                    250   

   Vehicle_Performance_High  Vehicle_Performance_Intermediate  \
0                         0                                 0   
1                         0                                 0   
2                         0                                 0   
3                         0                                 0   
4                         0                                 0   

   Vehicle_Anti_Theft_Device_Alarm Only  \
0                                     0   
1                                     0   
2                                     0   
3                                     0   
4                                     0   

   Vehicle_Anti_Theft_Device_Passive Disabling-Vehicle Recovery  \
0                                                  0              
1                                                  0              
2                                                  1              
3                                                  1              
4                                                  1              

   Vehicle_Passive_restraint_Y  Policy_Installment_Term  \
0                            1                        6   
1                            0                        6   
2                            1                        6   
3                            1                        6   
4                            1                        6   

   Vehicle_Med_Pay_Limit     ...       \
0                   1000     ...        
1                   2000     ...        
2                  50000     ...        
3                   5000     ...        
4                   5000     ...        

   Vehicle_Youthful_Good_Student_Code_Eligible for Good Student Credit      \
0                                                  0                         
1                                                  0                         
2                                                  0                         
3                                                  0                         
4                                                  0                         

   Vehicle_Driver_Points  Policy_CompanyY  EEA_Multi_Auto_Policies_IndicatorY  \
0                      0                1                                   1   
1                      0                0                                   0   
2                      0                1                                   1   
3                      0                1                                   1   
4                      0                1                                   1   

   EEA_Liability_Coverage_Only_IndicatorY  EEA_Agency_TypeY  \
0                                       1                 1   
1                                       0                 0   
2                                       1                 1   
3                                       1                 1   
4                                       1                 1   

   EEA_Packaged_Policy_IndicatorY  EEA_Full_Coverage_IndicatorY  \
0                               1                             1   
1                               0                             0   
2                               1                             1   
3                               1                             1   
4                               1                             1   

   Annual_Premium  Loss_Amount  
0          320.12          0.0  
1          259.70          0.0  
2          613.74          0.0  
3  

In [5]:
updated_data = og_data[og_data['Loss_Amount'] > 0]

In [6]:
updated_data.shape

(18869, 30)

In [7]:
og_data = og_data[og_data['Loss_Amount'] == 0]

In [8]:
og_data.shape

(405562, 30)

## Dividing Original Dataframe into chunks of 500

In [9]:
max_rows = 478
m = 22
chunked_dataframes = []
while len(og_data) > max_rows:
    top = og_data[:max_rows]
    top = top.append(updated_data[:m])
    chunked_dataframes.append(top)
    og_data = og_data[max_rows:]
    updated_data = updated_data[m:]
else:
    top = (og_data)
    top = top.append(updated_data)
    chunked_dataframes.append(top)

## List of Dataframes

In [10]:
len(chunked_dataframes)

849

In [11]:
first_dataframe = chunked_dataframes[0]

In [12]:
first_dataframe.shape

(500, 30)

## Function to Compress Dataframe of 500 rows into one row and making a new dataframe of one row

In [13]:
def compress_dataFrame(first):
    for row in first:
        first[row] = first[row].mean()
    first_new = first[:1]
    return first_new

## Appending Dataframes of one row each into a list of Dataframes

In [14]:
merged_dataframes = []
for dataframe in chunked_dataframes:
    merged_dataframes.append(compress_dataFrame(dataframe))

In [15]:
len(merged_dataframes)

849

## Merging the list of 849 dataframes into one final dataframe

In [16]:
final_dataframe = pd.concat([df for df in merged_dataframes])

## Final Dataframe compressing data from 5 million rows to 850 rows

In [17]:
final_dataframe.shape

(849, 30)

In [18]:
final_dataframe.head(10)

Vehicle_Symbol  Vehicle_Make_Year  \
0             10.026           1995.700   
495           10.016           1995.504   
993            9.644           1995.330   
1488          10.578           1995.828   
1978          10.554           1996.218   
2473          10.324           1996.390   
2976          10.378           1996.482   
3471          10.164           1995.934   
3964          10.120           1995.564   
4453          10.340           1996.046   

      Vehicle_Collision_Coverage_Deductible  Vehicle_Performance_High  \
0                                   264.974                     0.006   
495                                 242.412                     0.004   
993                                 258.630                     0.002   
1488                                250.956                     0.012   
1978                                256.118                     0.012   
2473                                276.464                     0.006   
2976                                284.496                     0.008   
3471                                267.252                     0.002   
3964                                260.524                     0.006   
4453                                277.262                     0.014   

      Vehicle_Performance_Intermediate  Vehicle_Anti_Theft_Device_Alarm Only  \
0                                0.024                                 0.058   
495                              0.020                                 0.056   
993                              0.020                                 0.060   
1488                             0.030                                 0.066   
1978                             0.022                                 0.068   
2473                             0.026                                 0.066   
2976                             0.024                                 0.070   
3471                             0.010                                 0.060   
3964                             0.016                                 0.062   
4453                             0.020                                 0.050   

      Vehicle_Anti_Theft_Device_Passive Disabling-Vehicle Recovery  \
0                                                 0.254              
495                                               0.240              
993                                               0.254              
1488                                              0.256              
1978                                              0.268              
2473                                              0.278              
2976                                              0.268              
3471                                              0.274              
3964                                              0.262              
4453                                              0.270              

      Vehicle_Passive_restraint_Y  Policy_Installment_Term  \
0                           0.590                    6.132   
495                         0.620                    6.084   
993                         0.600                    6.084   
1488                        0.664                    6.216   
1978                        0.662                    6.228   
2473                        0.680                    6.180   
2976                        0.672                    6.156   
3471                        0.660                    6.132   
3964                        0.610                    6.120   
4453                        0.626                    6.192   

      Vehicle_Med_Pay_Limit     ...       \
0                  2979.950     ...        
495                2997.888     ...        
993                3137.930     ...        
1488               3051.890     ...        
1978               3121.904     ...        
2473               2921.916     ...        
2976               3053.936     ...        
3471               2713.918     ...      